### ISSUE TITLE
General optimizations...

### ISSUE THREAD

**Issue by linas** [MOZI-AI/annotation-scheme#85]
~~So, in `rna.scm` I see this: `(list (cog-outgoing-set ...` This is pointless, because the outgoing set is already a list Wrapping it in a list again does nothing at all except waste cpu time.~~

~~That block of code would also run more efficiently if the if-tests were outside of the `map` instead of inside of it.  That way, the if-test wold be performed only once, instead of hundreds (or millions) of times.~~

Comment no longer applies, the code is no longer structured like this.

**Comment by linas** [comment]
The `node-info` function returns a list that is always of length one.  Then `append` is used to concatenate this list onto another list.  If instead, it just returned a single atom, then `cons` could be used, for a minor performance boost.

**Comment by linas** [comment]
The `add-loc` function contains this: 
```
          (AndLink
            (MemberLink
              child
              parent)
```
The MemberLink is a constant, and does not alter search results. It can be removed.

**Comment by linas** [comment]
The file `UniProt2Reactome_PE_Pathway.txt.scm` contains structures like this:
```
(AndLink
(MemberLink
 (MoleculeNode "Uniprot:A0A075B6P5")
(ConceptNode "R-HSA-166663"))

(EvaluationLink
 (PredicateNode "has_location")
 (ListLink
 (MoleculeNode "Uniprot:A0A075B6P5")
(ConceptNode "extracellular region")))

)
```

I believe the AndLink serves no purpose whatsoever, and can be safely removed.  This will make loading slightly faster.

**Comment by linas** [comment]
File loading performance could be increased by writing, at the start of the file
```
(define loc (PredicateNode "has_location"))
```
and then using that: `(Evaluation loc (List  ...))`

Also doing this for each gene, pathway, protein would also speed things a little bit:
```
(define muniA0A075B6P5 (Molecule "Uniprot:A0A075B6P5"))
(define rhsa166663 (Concept "R-HSA-166663"))
```
By using such defines, you save a tiny amount of time to drop into the C++ code and atomspace, to perform the lookup. Also, the files get slightly smaller.

**Comment by linas** [comment]
Closing since #137 was merged, and #150 tracks the file-enhancement idea
